craft_application/git/_git_repo.py

```diff
@@ -91,6 +91,35 @@ def get_git_repo_type(path: Path) -> GitType:
     return GitType.INVALID
 
 
+def parse_describe(describe_str: str) -> str:
+    """Parse git describe string to get a human-readable version.
+
+    Examples (git describe -> parse_describe):
+    4.1.1-0-gad012482d -> 4.1.1
+    4.1.1-16-g2d8943dbc -> 4.1.1.post16+git2d8943dbc
+    curl-8_11_0-0-gb1ef0e1 -> curl-8_11_0
+
+    For shallow clones or repositories missing tags:
+    0ae7c04 -> 0ae7c04
```

Comment:
Is this referenced from this? https://github.com/canonical/snapcraft/blob/6b94548db638ddb17acf49b7340a2b801a1ab4e7/tools/version.py#L25-L30

Just wondering if you found a more official source of our versioning convention.

Reply:
Actually I used this one, but I couldn't find an `official` source of this documentation:
https://github.com/canonical/charmcraft/blob/main/tools/version.py#L23